### ISSUE TITLE
Make sure borders work on high contrast mode

### ISSUE THREAD

**Issue by andresgalante** [patternfly/patternfly-next#286]
On #268 @christianvogt spotted that using a shadow as a border might generate a problem  conforming with high contrast mode in windows

> borders go to the foreground color and box-shadows are dropped. Therefore you'd lose the outline with these changes. However I haven't tried lately with edge and windows 10.

This issue is to look if this problem exist and if it does change the shadow for a border

**Comment by jgiardino** [comment]
Here's a great video that @garrett had shared with us previously with a suggestion on how to address this: https://www.youtube.com/watch?v=Wq2-cZ-vZBc

To summarize what the presenter suggests:
> Wherever shapes of actionable or overlay elements are created using background color or box shadows, include a 1px solid transparent border so that a visible outline will display when using a high contrast theme (e.g. input fields, buttons, modals)

**Comment by mindreeper2420** [comment]
These are the current buttons examples in High Contrast on Windows 10/Edge:

Mode 1:
<img width="1102" alt="screen shot 2018-06-11 at 4 52 29 pm" src="https://user-images.githubusercontent.com/4032718/41256539-e59fa18a-6d97-11e8-936a-1269e5172096.png">

Mode 2:
<img width="1107" alt="screen shot 2018-06-11 at 4 53 17 pm" src="https://user-images.githubusercontent.com/4032718/41256580-01830554-6d98-11e8-9131-39d3945928c8.png">

**Comment by mindreeper2420** [comment]
This is what they look like with `border: 1px solid transparent;` set:

Mode 1:
<img width="1092" alt="screen shot 2018-06-11 at 5 04 04 pm" src="https://user-images.githubusercontent.com/4032718/41257050-78e2de3e-6d99-11e8-913f-5bce826f361e.png">

Mode 2:
<img width="1098" alt="screen shot 2018-06-11 at 5 02 56 pm" src="https://user-images.githubusercontent.com/4032718/41257017-5d27568e-6d99-11e8-84e5-e58d5077a93c.png">

**Comment by mindreeper2420** [comment]
Dropdowns with High Contrast and Greyscale:
<img width="1108" alt="screen shot 2018-06-11 at 5 13 44 pm" src="https://user-images.githubusercontent.com/4032718/41257441-d726a1e6-6d9a-11e8-8f87-a65375d84fd6.png">

Dropdowns with High Contrast without Greyscale:
<img width="1102" alt="screen shot 2018-06-11 at 5 13 20 pm" src="https://user-images.githubusercontent.com/4032718/41257415-c32cc53a-6d9a-11e8-998f-4cd2837ef945.png">

**Comment by mindreeper2420** [comment]
After applying `border: 1px solid transparent;` to both `.pf-c-dropdown__toggle` and `.pf-c-dropdown__menu`:

<img width="1104" alt="screen shot 2018-06-11 at 5 23 24 pm" src="https://user-images.githubusercontent.com/4032718/41257799-2b575b06-6d9c-11e8-9267-ae10ab9caf55.png">

<img width="1091" alt="screen shot 2018-06-11 at 5 21 51 pm" src="https://user-images.githubusercontent.com/4032718/41257805-2e591fc4-6d9c-11e8-83f7-ecfee4f00019.png">

**Comment by mindreeper2420** [comment]
@andresgalante thoughts on the above implementations? Am I missing anything that needs to be covered?

**Comment by jgiardino** [comment]
Since you ask, form inputs and modal box also use shadows instead of borders. I definitely think those should get a similar transparent border treatment.

**Comment by mindreeper2420** [comment]
@jgiardino do checkboxes/radio buttons have outlines by default? I can update the inputs if we want to cover those here, but I'm not sure if the modal should be another story or not?

**Comment by jgiardino** [comment]
Oh, I meant the form inputs where we visually create a border by using box shadows, like text inputs, and also the select menus.

I think the way the issue description is written that you could include box modals with your PR for this issue. Or we could keep this issue open until a PR is merged for box modals?

But we should definitely update any element that would normally have a border, but the border was replaced with a box shadow.

**Comment by mindreeper2420** [comment]
So I've noticed that when adding `border: 1px solid transparent;` to the dropdown menus, it actually applies a solid border to the element. It looks as if it is picking up the box-shadow in some way?

The inputs and buttons work as expected - no border appears, unless in High Contrast mode.
